pkg/cmd/extension/manager.go

```diff
@@ -249,6 +250,64 @@ func (m *Manager) installDir() string {
 	return filepath.Join(m.dataDir(), "extensions")
 }
 
+func (m *Manager) Create(name string) error {
+	exe, err := m.lookPath("git")
+	if err != nil {
+		return err
+	}
+
+	err = os.Mkdir(name, 0755)
+	if err != nil {
+		return err
+	}
+
+	initCmd := m.newCommand(exe, "init", "-b", "main", "--quiet", name)
+	err = initCmd.Run()
+	if err != nil {
+		return err
+	}
+
+	fileTmpl := heredoc.Docf(`
+		#!/bin/bash
+		set -e
+		echo "Hello %[1]s"
+
+		# Examples to help get started:
+
+		# Determine if an executable is in the PATH
+		# if [ "$(which go)" = "" ]; then
+		#   echo "go not found in PATH"
+		#   exit 1
+		# fi
+
+		# Pass arguments through to a new process
+		# exec gh issue list "$@" -R cli/cli
+
+		# Using the gh api command to retrieve and format information
+		# QUERY='
+		#   query($endCursor: String) {
+		#     viewer {
+		#       repositories(first: 100, after: $endCursor) {
+		#         nodes {
+		#           nameWithOwner
+		#           stargazerCount
+		#         }
+		#       }
+		#     }
+		#   }
+		# '
+		# TEMPLATE='
+		#   {{- range $repo := .data.viewer.repositories.nodes -}}
+		#     {{- printf "name: %[2]s - stargazers: %[3]s\n" $repo.nameWithOwner $repo.stargazerCount -}}
+		#   {{- end -}}
+		# '
+		# exec gh api graphql --template="${TEMPLATE}" --paginate -f query="${QUERY}"
```

Comment:
Nit: could query go first and template last? So that it follows the order of operations: the requests is made first, then the response is processed using the template
```suggestion
		# exec gh api graphql -f query="${QUERY}"  --paginate --template="${TEMPLATE}"
```